From e6ba3112249a699f276e285182d2915b297c2e7f Mon Sep 17 00:00:00 2001
From: Ye Li <ye.li@nxp.com>
Date: Tue, 19 May 2026 17:18:42 +0800
Subject: [PATCH 14/24] LF-18073 iMX94: Add flash_all_no_m33s target

Add flash_all_no_m33s target to exclude M33 sync image.
On EVK, M33s uses LPUART8 which uses the same pins for JTAG, so
users cannot attach with JTAG. Add this target to build flash.bin
without M33s to ease the debug process.

Upstream-Status: Pending

Signed-off-by: Ye Li <ye.li@nxp.com>
---
 iMX94/soc.mak | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/iMX94/soc.mak b/iMX94/soc.mak
index 1321be8..d8fcd9e 100644
--- a/iMX94/soc.mak
+++ b/iMX94/soc.mak
@@ -581,6 +581,14 @@ flash_all_no_m7: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M33S_IMG) u-boot-atf-containe
 		   $(V2X_DUMMY) -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) -out flash.bin
 	$(call append_container,u-boot-atf-container.img,1)
 
+flash_all_no_m33s: $(MKIMG) $(AHAB_IMG) $(MCU_IMG) $(M70_IMG) $(M71_IMG) u-boot-atf-container.img $(SPL_A55_IMG) $(OEI_IMG_M33)
+	./$(MKIMG) -soc IMX9 -cntr_version 2 $(MMC_FAST_HASH) -append $(AHAB_IMG) -c $(OEI_OPT_M33) -msel $(MSEL) \
+		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
+		   -m7 $(M70_IMG) 0 $(M70_TCM_ADDR) $(M70_TCM_ADDR_ALIAS)  \
+		   -m7 $(M71_IMG) 1 $(M71_TCM_ADDR) $(M71_TCM_ADDR_ALIAS) \
+		   $(V2X_DUMMY) -ap $(SPL_A55_IMG) a55 $(SPL_LOAD_ADDR_M33_VIEW) -out flash.bin
+	$(call append_container,u-boot-atf-container.img,1)
+
 flash_all_no_ahabfw: $(MKIMG) $(MCU_IMG) $(M33S_IMG) $(M70_IMG) $(M71_IMG) u-boot-atf-container.img $(SPL_A55_IMG) $(OEI_IMG_M33)
 	./$(MKIMG) -soc IMX9 -cntr_version 2 $(CNTR_FLAG) -c $(OEI_OPT_M33) -msel $(MSEL) \
 		   -m33 $(MCU_IMG) 0 $(MCU_TCM_ADDR) \
